pkg/chunk/table_manager.go

```diff
@@ -94,6 +96,34 @@ type TableManagerConfig struct {
 	ChunkTables ProvisionConfig `yaml:"chunk_tables_provisioning"`
 }
 
+// UnmarshalYAML implements the yaml.Unmarshaler interface. To support RetentionPeriod.
+func (cfg *TableManagerConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&cfg); err != nil {
+		return err
+	}
+
+	if cfg.RetentionPeriodModel > 0 {
+		cfg.RetentionPeriod = time.Duration(cfg.RetentionPeriodModel)
+	}
+
+	return nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface. To support RetentionPeriod.
+func (cfg *TableManagerConfig) MarshalYAML() (interface{}, error) {
+	cfg.RetentionPeriodModel = model.Duration(cfg.RetentionPeriod)
+	return cfg, nil
+}
+
+// Validate validates the config.
+func (cfg *TableManagerConfig) Validate() error {
+	if cfg.RetentionPeriodModel > 0 {
+		cfg.RetentionPeriod = time.Duration(cfg.RetentionPeriodModel)
+	}
```

Comment:
Why do you need this? Aren't you already covered by the custom marshalling? I'm asking because having business logic here breaks the `Validate()` contract which is expected to not manipulate the config and just return an error if it's valid.

Reply:
This is for flags. I don't know of a good way to set the right flag.

Reply:
Right. The only idea I have is to add a custom data type for the flag and having the `Set()` writing to both `cfg.RetentionPeriodModel` and `cfg.RetentionPeriod` (this one copying the value from the model) but it adds extra complexity and it's not worth to me.

So, I would just suggest you to add a comment before this `if` to explain why we do it (I didn't get it at first).